src/main/kotlin/no/skatteetaten/aurora/boober/platform/AbstractHandler.kt

```diff
@@ -0,0 +1,27 @@
+package no.skatteetaten.aurora.boober.platform
+
+import no.skatteetaten.aurora.boober.mapper.AuroraConfigFieldHandler
+import no.skatteetaten.aurora.boober.mapper.AuroraConfigFields
+import no.skatteetaten.aurora.boober.model.ApplicationId
+import no.skatteetaten.aurora.boober.model.AuroraConfigFile
+import no.skatteetaten.aurora.boober.model.Container
+
+abstract class AbstractHandler {
+    open fun handlers(handlers: Set<AuroraConfigFieldHandler>): Set<AuroraConfigFieldHandler> = handlers
+    /*
+     * This method should throw a AuroraConfigException on errors
+     */
+    open fun validate(applicationId: ApplicationId,
+                      applicationFiles: List<AuroraConfigFile>,
+                      handlers: Set<AuroraConfigFieldHandler>,
+                      auroraConfigFields: AuroraConfigFields) {
+    }
+
+    abstract val container: List<Container>
+}
+
+//TODO: Is there an easier typesafe way to instantiate these classes?
+enum class ApplicationPlatform(val handler: AbstractHandler) {
```

Comment:
This should probably not be declared here as it creates a "circular dependency" between the AbstractHandler-file and the implementations.

Reply:
The intent here was to have one place to see what all the implementations are. Ideally I would like not to have the enum at all and just map applicationPlatform straight to an handler but I am not sure if that can be done in a elegant way without reflection.